File: variants/java-gradle/aibest/src/main/java/net/pijcke/nim/aibest/BestAIFactory.java
package net.pijcke.nim.aibest;

import net.pijcke.nim.rules.Player;
import net.pijcke.nim.rules.PlayerFactory;

public class BestAIFactory implements PlayerFactory {

    @Override
    public String label() {
        return "AI : Best";
    }

    @Override
    public Player build() {
        return new BestAI();
    }

}

File: variants/kotlin-gradle/nim/aibest/src/main/kotlin/net/pijcke/nim/aibest/BestAiFactory.kt
package net.pijcke.nim.aibest

import net.pijcke.nim.rules.Player
import net.pijcke.nim.rules.PlayerFactory

class BestAiFactory : PlayerFactory {

    override fun label() = "AI : Best"

    override fun build(): Player {
        return BestAi()
    }

}